hazelcast/src/main/java/com/hazelcast/collection/impl/list/ListProxyImpl.java

```diff
@@ -130,15 +132,15 @@ public boolean addAll(int index, Collection<? extends E> c) {
 
     @Override
     public List<E> subList(int fromIndex, int toIndex) {
-        final ListSubOperation operation = new ListSubOperation(name, fromIndex, toIndex);
-        final SerializableCollection result = invoke(operation);
-        final Collection<Data> collection = result.getCollection();
-        final List<E> list = new ArrayList<E>(collection.size());
+        ListSubOperation operation = new ListSubOperation(name, fromIndex, toIndex);
+        SerializableCollection result = invoke(operation);
+        Collection<Data> collection = result.getCollection();
```

Comment:
This is what I removed for the last performance improvements. We might just want to create a view wrapper without copying the content whereas in this case it is not as bad since we don't copy hashmaps / hashsets which are pretty bad due to re-hashing.